packages/create-amplify/src/package-manager-controller/pnpm_package_manager_controller.ts

```diff
@@ -0,0 +1,82 @@
+import { existsSync as _existsSync } from 'fs';
+import { execa as _execa } from 'execa';
+import { executeWithDebugLogger } from '../execute_with_logger.js';
+import {
+  DependencyType,
+  PackageManagerController,
+  PackageManagerControllerFactory,
+} from './package_manager_controller_factory.js';
+
+/**
+ *
+ */
+export class PnpmPackageManagerController implements PackageManagerController {
+  private readonly execa = _execa;
+  private readonly packageManagerProps = {
+    name: 'pnpm',
+    executable: 'pnpm',
+    binaryRunner: 'pnpm',
+    installCommand: 'add',
+    lockFile: 'pnpm-lock.yaml',
+    initDefault: ['init'],
+  };
+
+  /**
+   * Abstraction around pnpm commands that are needed to initialize a project and install dependencies
+   */
+  constructor(
+    private readonly projectRoot: string,
+    private readonly packageManagerControllerFactory: PackageManagerControllerFactory
+  ) {}
+
+  /**
+   * Installs the given package names as devDependencies
+   */
+  installDependencies = async (
+    packageNames: string[],
+    type: DependencyType
+  ): Promise<void> => {
+    const args = [`${this.packageManagerProps.installCommand}`].concat(
+      ...packageNames
+    );
+    if (type === 'dev') {
+      args.push('-D');
+    }
+
+    await executeWithDebugLogger(
+      this.projectRoot,
+      this.packageManagerProps.executable,
+      args,
+      this.execa
+    );
+  };
+
+  ensureInitialized = async () => {
+    await this.packageManagerControllerFactory.initializeProject(
+      this.packageManagerProps
+    );
+  };
+
+  getWelcomeMessage = () => {
+    const cdCommand =
+      process.cwd() === this.projectRoot
+        ? ''
+        : `cd .${this.projectRoot.replace(process.cwd(), '')}; `;
+
+    return `Welcome to AWS Amplify! 
+Run \`${this.packageManagerProps.binaryRunner} amplify help\` for a list of available commands. 
+Get started by running \`${cdCommand}${this.packageManagerProps.binaryRunner} amplify sandbox\`.`;
```

Comment:
We can have the PackageManagerController only compute this part of the string. The rest can be computed in the AmplifyProjectCreator

Reply:
> this part of the string

Which part?

Reply:
Done